### PR TITLE
Support marketplace purchase event

### PR DESCRIPTION
**Pull request by AlistairB**
**Issue reference:**

Closes https://github.com/cuedo/github-webhooks/issues/46

**Submission Checklist:**
* [x] Have you followed the guidelines in our [Contributing](../../CONTRIBUTING.md) document (for example, is your tree a clean merge)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission build?
* [x] Does your submission pass tests?
* [x] Have you run lints on your code locally prior to submission?
* [x] Have you updated *all* of the cabal/nix infrastructure?
* [x] Is this a breaking change? Have you discussed this?

### CODE REVIEW

File: src/GitHub/Data/Webhooks/Payload.hs [hunk 1]
```diff
@@ -69,6 +72,7 @@ import           Data.Text                (Text)
 import qualified Data.Text                as T
 import           Data.Vector              (Vector)
 import           GHC.Generics             (Generic)
+import Data.Vector.Fusion.Util (Id)
```
Comment (kvanbere) [inline src/GitHub/Data/Webhooks/Payload.hs:75]: What is this used for?

**Comment by kvanbere** [comment]
Thanks for the PR, happy to merge once the implementation is complete and `marketplacePurchaseEventFixture` is fixed to a valid fixture result :)

File: src/GitHub/Data/Webhooks/Payload.hs (hunk 1 above)
Comment (AlistairB) [reply src/GitHub/Data/Webhooks/Payload.hs:75]: Oh oops. The PR is very WIP, I'll let you know once it is ready for review. Thanks :)

**Comment by AlistairB** [comment]
@kvanbere should be good now! Let me know if its need any changes.

**Review by kvanbere — APPROVED** [review]
Thank you very much for contributing high quality code.

**Comment by kvanbere** [comment]
I will make a `0.16.0` release a priority. There has not been a new release in some time, and it would be good to get these changes on Stackage.

**Pull request merged**